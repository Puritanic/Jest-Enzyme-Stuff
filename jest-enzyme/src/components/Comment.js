import React from 'react';

const Comment = ({ comment }) => {
	return (
		<ul className="comment">
			<p className="comment__content">{comment.content}</p>
			<small className="comment__author">{comment.author}</small>
		</ul>
	);
};

export default Comment;
