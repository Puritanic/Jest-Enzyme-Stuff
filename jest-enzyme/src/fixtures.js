export const comments = [
	{ id: 67829, content: 'Just a comment', author: 'PuritanicC' },
	{ id: 70501, content: 'Just a comment', author: 'PuritanicA' },
	{ id: 19898, content: 'Just a comment', author: 'PuritanicB' },
];
