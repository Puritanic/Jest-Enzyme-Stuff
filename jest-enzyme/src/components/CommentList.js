import React from 'react';

import Comment from './Comment';

const CommentList = props => {
	return (
		<ul className="comment__list">
			{props.comments &&
				props.comments.map(comment => <Comment key={comment.id} comment={comment} />)}
		</ul>
	);
};

export default CommentList;
