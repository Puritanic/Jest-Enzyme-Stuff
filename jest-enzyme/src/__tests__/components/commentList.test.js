/* eslint-disable no-undef */
import React from 'react';

import { comments } from '../../fixtures';
import CommentList from '../../components/CommentList';
import Comment from '../../components/Comment';

describe('<CommentList />', () => {
	let wrapper;

	beforeEach(() => {
		wrapper = shallow(<CommentList comments={comments} />);
	});

	it('should match snapshot', () => {
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('should render all comments available', () => {
		expect(wrapper.children().length).toEqual(comments.length);
	});

	it('should render <Comment /> for each comment', () => {
		expect(wrapper.find(Comment).length).toBe(comments.length);
	});
});
