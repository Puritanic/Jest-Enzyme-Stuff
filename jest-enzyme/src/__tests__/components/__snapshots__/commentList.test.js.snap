// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentList /> should match snapshot 1`] = `
<ul
  className="comment__list"
>
  <Comment
    comment={
      Object {
        "author": "PuritanicC",
        "content": "Just a comment",
        "id": 67829,
      }
    }
    key="67829"
  />
  <Comment
    comment={
      Object {
        "author": "PuritanicA",
        "content": "Just a comment",
        "id": 70501,
      }
    }
    key="70501"
  />
  <Comment
    comment={
      Object {
        "author": "PuritanicB",
        "content": "Just a comment",
        "id": 19898,
      }
    }
    key="19898"
  />
</ul>
`;
